### DISCUSSION TITLE
3. ROI 성숙도 등급 판단 (CNN)

### DISCUSSION THREAD

**Discussion by SHEYCHAE** [kikiru328/Bone_Detection#24] [Study]
# 3. ROI 성숙도 등급 판단 (CNN)
VGG BA NET 사용!
- VGG 입력 사이즈 224*224 -> ROI크기가 주로 100*100 -> 입력영상사이즈 112*112로 받을 수 있도록 구조 수정
- 3채널 영상을 사용할 필요없어서 1채널(그레이스케일) 영상을 입력영상으로 받음 
- depth크기 8배 감소 
- Convolution Layer 두개 Fully Connected Layer 하나를 제거해 모델 복잡성 최소화 및 학습의 과적합 방지
- hyperparameter: Learning rate 0.001, Loss function -CrossEntropy, Batch-256, Epoch-30 
- 각 Convolution Layer 3*3의 Kernel과 1의 Stride 사용 
- Pooling Layer는 모두 Max Pooling 방식으로 2*2의 Kernel과 2의 Stride 사용, Zero Padding 사용
- Activation Function :Relu 사용 
- Dropout은 첫 Fully Connected Layer 후에 한 번 이루어지며 최종적인 뼈의 등급은 Softmax를 통해 분류 

- 학습을 10 Fold Cross Validation방식으로 진행하니까
  전체 학습 데이터 집합을 1:9 비율로 ㄴ누어 서로다른 학습데이터 집합 10개 생성
  -> 학습 진행시 하나의 학습데이터 집합으로부터 10개의 모델 생성 
- 골성숙도 등급 분류를 위해 실험영상을 입력하고 실험진행시 결과물로 A~I등급중 어느 등급에 해당되는지 확률정보 생성
- 10개의 결과가 나옴 - Ensemble적용하여 최종 결과 얻음
- 10개의 결과물의 각 등급에 대한 확률에 평균을 취한 뒤 최종적으로 제일 높은 확률을 띄는 등급이 해당 뼈 부위의 성숙도 등급이 됨

- 13개 뼈의 성숙도 등급이 모두 정해지면 실험으로 얻어낸 각 등급들의 점수를 총합하여 RUS Score에 따른 골연령표와 비교하여 환자의 골연령 결정
